Tests/MongoSwiftTests/DNSSeedlistTests.swift

```diff
@@ -65,17 +65,29 @@ final class DNSSeedlistTests: MongoSwiftTestCase {
 
             // TODO: DRIVERS-796 or DRIVERS-990: unskip this test
             guard fileName != "txt-record-with-overridden-uri-option.json" else {
-                return
+                continue
             }
 
             // Enclose all of the potentially throwing code in `doTest`. Sometimes the expected errors come when
             // parsing the URI, and other times they are not until we try to select a server.
             func doTest() throws -> ConnectionString {
-                let opts = MongoClientOptions(
+                var opts = MongoClientOptions(
                     tlsAllowInvalidCertificates: true,
                     tlsCAFile: URL(string: MongoSwiftTestCase.sslCAFilePath ?? ""),
                     tlsCertificateKeyFile: URL(string: MongoSwiftTestCase.sslPEMKeyFilePath ?? "")
                 )
+
+                // NOTE: since we have to connect to the replica set and attempt server selection to get the SRV
```

Comment:
here's why we didn't see this before:

Before, the URI had ssl=false, but the client pool had the TLS opts set above (CA file etc.). The client pool having those options set on it let to libmongoc inferring that to mean SSL=true for the pool, which overrode the URI's ssl=false.

Now that we set everything on the URI though, the URI has explicit SSL=false plus the options above. In that case it seems the ssl=false wins out, and no override occurs.

Reply:
This seems like we may be sidestepping the purpose of the test though. Aren't we supposed to be verifying that srv resolution works even if the user explicitly disables ssl?

Reply:
so I copied this from what libmongoc does in their equivalent tests:
https://github.com/mongodb/mongo-c-driver/blob/49d5590b0164e629c6adc1ed1b88e6852f62bba9/src/libmongoc/tests/test-mongoc-dns.c#L210

in other drivers, it looks like they are able to implement this test by not actually sending commands and just parsing the connection string and resolving the SRV and TXT records. 

for example, Node implements these tests via a `parseConnectionString` method: https://github.com/mongodb/node-mongodb-native/blob/5808e12f13e39bfddb26cc15fe9b4dd751e1f363/test/unit/core/mongodb_srv.test.js#L32

and it looks like yall do the same in Rust?
https://github.com/mongodb/mongo-rust-driver/blob/a497db1150082fba85880d6b5a652ce8a78932c4/src/test/spec/initial_dns_seedlist_discovery.rs#L40

so I think what we are doing here with checking `tempConnStr.options?["tls"]` is basically equivalent to the Node and Rust tests, the only difference is that we do it prior to SRV/TXT record resolution. note that the SSL/TLS option can not be set via TXT record per the spec though, so it's not possible that this option would change from doing resolution.

taking the route here of overriding the ssl option allows us to make the host list assertions for this test successfully, as doing that requires us to successfully start SDAM and inspect the resulting topology events. an alternative route would be to not override the option, but skip the host list assertion for this case. 

cc @kevinAlbs in case he has any insight

Reply:
I think the point of the test is to verify that users can override the default SSL=true behavior that they get when using a mongodb+srv URI.
we could alternatively maybe only run this particular test case when SSL is turned off for the test replica set? I think that would allow us to confirm that the ssl=false option is successfully respected.

Reply:
I think the drivers that just test the parsing are likely not run the tests the way they're supposed to be run. For instance, it looks like the Rust runner is just ignoring the `hosts` field of the spec test file completely.

>we could alternatively maybe only run this particular test case when SSL is turned off for the test replica set? I think that would allow us to confirm that the ssl=false option is successfully respected.

This seems like the right path.  As the spec is written right now, it seems to have conflicting requirements because it says both that the replica set MUST be started with ssl but also that drivers should be able to do an SDAM pass with the given options. Something like a `runOn` with `ssl: true` would be useful to include in the spec test format for stuff like this.

Reply:
I don't have much additional insight, but I agree it seems like the spec has conflicting requirements. Perhaps this warrants a spec change. Since I don't think all drivers can set SSL options after client construction from a URI, maybe the test should omit the list of hosts?

IMO following suit with libmongoc's tests seem reasonable to me, but so does Patrick's suggestion. Either way you'll need a special case for this test.

Reply:
I've updated to run this test without SSL now. I also opened [DRIVERS-1324](https://jira.mongodb.org/browse/DRIVERS-1324) regarding the issues with the spec tests.